### ISSUE TITLE
`prqlc compile` is not working as expected

### ISSUE THREAD

**Issue by eitsupi** [PRQL/prql#2581]
### What happened?

After #2570, the `prqlc compile` command no longer seems to work as expected.
Perhaps we need to undo the changes made to this command?

For example, download a binary from https://github.com/PRQL/prql/actions/runs/4991728868 and run the example command:

```sh
❯ echo "from employees | filter has_dog | select salary" | ./prqlc.exe compile
[E0001] Error: Missing main pipeline
```

https://github.com/PRQL/prql/blob/6d02d8704d608d01b73681459369ac6124418fad/prql-compiler/prqlc/README.md?plain=1#L23-L34

Interactive mode does not work either.

```sh
❯ ./prqlc.exe compile
Enter PRQL, then ctrl-d:

from a
[E0001] Error: Missing main pipeline
```

### PRQL input

```elm
NA
```

### SQL output

```SQL
NA
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by max-sixty** [comment]
Thanks for spotting, this indeed is a bad bug! I'll have a look later unless someone gets there first.

**Comment by aljazerzen** [comment]
Ah, I know what's the problem:

I've added a third param to the IOArgs:
```
prqlc compile input.prql output.prql my_query
```
... where `my_query` is the fq name of the variable that should be compiled to SQL.

I guess that the logic to pick appropriate default main pipeline is not working correctly.

**Comment by eitsupi** [comment]
Thank you for the fix!
Definitely need to enhance the testing against the CLI like #2586 to prevent regression...

**Comment by max-sixty** [comment]
FWIW I don't think this is quite solved — the test in https://github.com/prql/prql/blob/2fe0176d0843a6d6d8508c8f7f0cd7765e37acce/prql-compiler/prqlc/tests/test.rs#L69 is still broken.

I can have a look later if no one gets there first!

**Comment by aljazerzen** [comment]
The problem is that `insta-cmd` will not pass stdin when an arg is applied. Yes, it's that bizarre.

**Comment by max-sixty** [comment]
Oh gosh, you're right! Sorry, I hadn't thought to check that! I'll have a look at that but we can def close this...

**Comment by whoarusty** [comment]
I was able to get the Windows binary of prqlc -- thanks @eitsupi -- and tried to run the example, but got a different seemingly error than the one above.

```
>prqlc -V
prqlc 0.8.1

>prqlc compile
Enter PRQL, then press ctrl-z to compile:

"from employees | select {name, age}"Error:
   ╭─[Root.prql:1:1]
   │
 1 │ "from employees | select {name, age}"
   │ ──────────────────┬──────────────────
   │                   ╰──────────────────── main expected type `relation`, but found type `text`
   │
   │ Help: Type `relation` expands to `[tuple_of_scalars]` 
```

Does this have to do with the pipes in the PRQL query?

If I don't use the pipes, it seems fine:

```
>prqlc compile
Enter PRQL, then press ctrl-z to compile:

from employees
select {name, age}

SELECT
  name,
  age
FROM
  employees

-- Generated by PRQL compiler version:0.8.1 (https://prql-lang.org)
```

**Comment by eitsupi** [comment]
@whoarusty Didn't you enclose your query in `"`?

**Comment by whoarusty** [comment]
Oh, OK -- I see.

If I don't use the " in the interactive mode it works. I was used to prql-query where I had to use them.

```
> prqlc compile
Enter PRQL, then press ctrl-z to compile:

from employees | filter has_dog | select salary
SELECT
  salary
FROM
  employees
WHERE
  has_dog

-- Generated by PRQL compiler version:0.8.1 (https://prql-lang.org)
```

**Comment by eitsupi** [comment]
I'm glad it worked!

Please let us know if you find any bugs (obviously they are rarely tested!).
